app/views/TeamInfoPage.scala.html

```diff
@@ -171,6 +167,29 @@ <h3 class="board__heading">Team Members</h3>
     @dependenciesTeamPartial(teamName, masterTeamDependencies, prodDependencies, showLink = true)
 }
 
+@addSlackInfo(label: String, slackInfo: SlackInfo) = {
+  @slackInfo match {
+    case SlackInfo(_, url) if !(url.startsWith("http://") || url.startsWith("https://"))  => {
```

Comment:
can we move these filters into `SlackInfo`?
Maybe `SlackInfo.hasValidUrl`.
Similarly `SlackInfo.hasValidName`?  (the regex could also be cached)

Could even model with something like `SlackInfo = SlackInfo(name, url) | InvalidUrl(name) | InvalidName(url)`?

Reply:
Thanks, moved as suggested and created some unit tests.